modules/polaris/src/main/ts/ephox/polaris/words/Words.ts

```diff
@@ -38,25 +39,57 @@ export interface WordsWithIndices<T> {
   readonly indices: WordIndex[];
 }
 
+// const isAnAcronym = (str: string) => new RegExp(/^([A-Z]\.)+$/).test(str);
+
+interface CouldBeAnAcronym {
+  prev: undefined | number | string;
+  check: undefined | boolean;
+  isTheLast: boolean;
+}
+
 const findWordsWithIndices = <T>(chars: Word<T>, sChars: string[], characterMap: CharacterMap, options: WordOptions): WordsWithIndices<T> => {
   const words: Word<T>[] = [];
   const indices: WordIndex[] = [];
   let word: Word<T> = [];
 
+  const couldBeAnAcronym: CouldBeAnAcronym = {
+    prev: undefined,
+    check: undefined,
+    isTheLast: false
+  };
+
   // Loop through each character in the classification map and determine whether
   // it precedes a word boundary, building an array of distinct words as we go.
   for (let i = 0; i < characterMap.length; ++i) {
 
     // Append this character to the current word.
     word.push(chars[i]);
 
+    const onWordBoundary = isWordBoundary(characterMap, i);
+    const dotAfterLetter = sChars[i] === '.' && couldBeAnAcronym.prev === ci.ALETTER;
+    const letterAfterDot = characterMap[i] === ci.ALETTER && (couldBeAnAcronym.prev === '.' || couldBeAnAcronym.prev === undefined);
+    if (couldBeAnAcronym.check || couldBeAnAcronym.check === undefined) {
+      if (dotAfterLetter) {
+        couldBeAnAcronym.prev = sChars[i];
+        couldBeAnAcronym.check = true;
+        if (onWordBoundary) {
+          couldBeAnAcronym.isTheLast = true;
+        }
+      } else if (letterAfterDot && sChars[i + 1] === '.') {
+        couldBeAnAcronym.prev = characterMap[i];
+        couldBeAnAcronym.check = true;
+      } else {
+        couldBeAnAcronym.check = false;
+      }
+    }
     // If there's a word boundary between the current character and the next character,
+    // (and this boundary doesn't depend from a dot at the end of an acronym)
     // append the current word to the words array and start building a new word.
-    if (isWordBoundary(characterMap, i)) {
+    if (onWordBoundary && (couldBeAnAcronym.isTheLast || !couldBeAnAcronym.check)) {
```

Comment:
Since `isTheLast` is only set to true when `onWordBoundary` is true, and all this code only runs when `onWordBoundary` is true why does `isTheLast` exist? Looking at how it's set, it could probably be replaced with a check for `dotAfterLetter`.

But please make sure your tests capture the edge cases so they can confirm whether I'm wrong.

With this and the `prev` change your mutable additions collapse to a `check` boolean.